zebra-chain/src/block.rs

```diff
@@ -86,6 +86,34 @@ impl Block {
             Some(height) => Commitment::from_bytes(self.header.commitment_bytes, network, height),
         }
     }
+
+    /// Check if the `network_upgrade` fields from the transactions match block height.
+    ///
+    /// # Consensus rule:
+    ///
+    ///  The nConsensusBranchId field MUST match the consensus branch ID used for
+    ///  SIGHASH transaction hashes, as specifed in [ZIP-244] ([7.1]).
+    ///
+    /// [ZIP-244]: https://zips.z.cash/zip-0244
+    /// [7.1]: https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus
+    pub fn check_transaction_network_upgrade(
```

Comment:
Maybe a more specific name:
```suggestion
    pub fn check_consensus_branch_id_consistency(
```

Reply:
Yea, i am ok with this, i changed it at https://github.com/ZcashFoundation/zebra/pull/2354/commits/ee21e3b5145f4afc54508dfe609a7e02a14b4e13